### PR TITLE
Fix for the attchmentIndex

### PR DESCRIPTION
**Pull request by cyastella**
### Summary

fix the attachmentIndex since when it is int type, it can't be change to the omitempty. Thus we change it to the int pointer type.
### Implementation details

change the attachmentIndex to pointer.

- In awsvpc mode, it points to zero value, 

- in bridge/host mode, it become nil.

### Testing

New tests cover the changes:  yes

### Licensing

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by cyastella** [comment]
the build is breaking because the `aws:feature/v4_metadata` branch is out of sync with `aws:dev`; to pass the CI tests, will need to merge in `HEAD` from `aws:dev` branch.

**Comment by cyastella** [comment]
> the build is breaking because the `aws:feature/v4_metadata` branch is out of sync with `aws:dev`; to pass the CI tests, will need to merge in `HEAD` from `aws:dev` branch.
> 
> created #2393 to run the CI tests with the latest changes from dev

#2393 passed all CI checks; will rebase `aws:feature/v4_metadata` to `HEAD` of `aws:dev` after this change is merged

File: agent/handlers/v4/response.go [hunk 1]
```diff
@@ -172,11 +172,18 @@ func newNetworkInterfaceProperties(task *apitask.Task) (NetworkInterfaceProperti
 			"v4 metadata response: unable to parse subnet ipv4 address '%s'",
 			eni.SubnetGatewayIPV4Address)
 	}
+
+	var attachmentIndexPtr *int = nil
```
Comment (fenxiong) [inline agent/handlers/v4/response.go]: "= nil" is not needed. it's nil by default

Comment (cyastella) [reply agent/handlers/v4/response.go]: sure, will change it.

**Pull request merged**